preload_data.sh

```diff
@@ -1815,22 +1822,24 @@ function deletemongocopy {
     error "Cannot switch to $TO_NAMESPACE"
   fi
 
-  #delete all other resources EXCEPT icp-mongodb-admin
-  ${OC} delete statefulset icp-mongodb --ignore-not-found
-  ${OC} delete service icp-mongodb --ignore-not-found
-  ${OC} delete issuer god-issuer --ignore-not-found
-  ${OC} delete cm ibm-cpp-config --ignore-not-found
-  ${OC} delete certificate icp-mongodb-client-cert --ignore-not-found
-  ${OC} delete cm icp-mongodb --ignore-not-found
-  ${OC} delete cm icp-mongodb-init --ignore-not-found
-  ${OC} delete cm icp-mongodb-install --ignore-not-found
-  ${OC} delete secret icp-mongodb-keyfile --ignore-not-found
-  ${OC} delete secret icp-mongodb-metrics --ignore-not-found
-  ${OC} delete sa ibm-mongodb-operand --ignore-not-found
-  ${OC} delete service mongodb --ignore-not-found
-  ${OC} delete certificate mongodb-root-ca-cert --ignore-not-found
-  ${OC} delete issuer mongodb-root-ca-issuer --ignore-not-found
-  ${OC} delete cm namespace-scope --ignore-not-found
+  # delete all other resources EXCEPT icp-mongodb-admin
+  # Keep all necessary resources for the original mongo instance,
+  # this is the change for MongoDB to EDB migration.
+  # ${OC} delete statefulset icp-mongodb --ignore-not-found
+  # ${OC} delete service icp-mongodb --ignore-not-found
+  # ${OC} delete issuer god-issuer --ignore-not-found
+  # ${OC} delete cm ibm-cpp-config --ignore-not-found
+  # ${OC} delete certificate icp-mongodb-client-cert --ignore-not-found
+  # ${OC} delete cm icp-mongodb --ignore-not-found
+  # ${OC} delete cm icp-mongodb-init --ignore-not-found
+  # ${OC} delete cm icp-mongodb-install --ignore-not-found
+  # ${OC} delete secret icp-mongodb-keyfile --ignore-not-found
+  # ${OC} delete secret icp-mongodb-metrics --ignore-not-found
+  # ${OC} delete sa ibm-mongodb-operand --ignore-not-found
+  # ${OC} delete service mongodb --ignore-not-found
+  # ${OC} delete certificate mongodb-root-ca-cert --ignore-not-found
+  # ${OC} delete issuer mongodb-root-ca-issuer --ignore-not-found
+  # ${OC} delete cm namespace-scope --ignore-not-found
```

Comment:
Do we still want to include the deletion steps somewhere else? Or here with some condition to trigger?

Reply:
Oh yeah, this is a good point, Ben!

I will introduce a new flag `-c, --clean-mode=true`, if user passes the parameter, it will delete those resources.
Please let me know if you have any other thoughts. Thank you!